prog.go

```diff
@@ -92,6 +92,10 @@ type ProgramSpec struct {
 
 	// The byte order this program was compiled for, may be nil.
 	ByteOrder binary.ByteOrder
+
+	// Programs called by this ProgramSpec's bytecode. Populated during ELF
+	// loading and read by the linker during marshaling.
+	neighbours []*ProgramSpec
```

Comment:
It would be really cool if we could expose this somehow, that would let users:

* Generate eBPF functions in assembly and link them with C code by adding a `neighbour`.
* Change which functions are linked in. For example one C program could be linked with different implementations of a function by changing the `neighbours` in the spec (although that might require renaming functions, which would ruin the BTF).
That could also let users have a common "library" ELF what is linked with their programs.

But maybe a `ProgramSpec.Link(o ProgramSpec)` to link one program and it's neighbors to another would be enough?

Reply:
I don't think simply exporting this would work very well, but we could add a feature down the line for adding or replacing ProgramSpecs and re-running the linking/reference discovery step. In fact, the work in this PR is even necessary to make something like that work properly, since linking is now no longer a 'monolithic' process.

Can we revisit this later, preferably with some concrete use cases?